### PR TITLE
chore: hoist workspace packages

### PR DESCRIPTION
**Pull request by tmastrom**
Hoist the following packages to the workspace level 
- tsc-alias
- vite-node 
- vue-tsc
- eslint
- @typescript-eslint/eslint-plugin
- tsconfig-paths 
- nodemon
- ts-node

### CODE REVIEW

**Comment by hanspagel** [comment]
I wonder if we can still run the type check in a package folder, though? I do this a lot. If not, there’s probably a workaround.

**Comment by tmastrom** [comment]
> I wonder if we can still run the type check in a package folder, though? I do this a lot. If not, there’s probably a workaround.

I'm still able to run `pnpm types:build` and `pnpm types:check` inside of a package folder. You may need to have typescript installed globally on your machine to make this possible.

File: examples/fastify-api-reference/package.json [hunk 1]
```diff
@@ -22,9 +22,6 @@
     "fastify": "^4.26.2"
   },
   "devDependencies": {
-    "nodemon": "^3.1.3",
-    "tsc-alias": "^1.8.8",
-    "vite": "^5.2.10",
-    "vite-node": "^1.3.1"
+    "vite": "^5.2.10"
```
Comment (hanspagel) [inline examples/fastify-api-reference/package.json:25]: Do we have to keep Vite, or can we move this to the root package.json, too? 🤔

**Review by hanspagel — APPROVED** [review]
So fresh, so clean, love it!

**Comment by geoffgscott** [comment]
Looks good just need to resolve conflicts

**Comment by tmastrom** [comment]
Closing in favour of https://github.com/scalar/scalar/pull/2524 to avoid rebase hell

**Pull request closed without merging**